package.json

```diff
@@ -24,11 +24,17 @@
     "cli-color": "^2.0.2",
     "fs-extra": "^10.1.0",
     "mocha": "^8.3.2",
+    "node-gyp": "9.1.0",
```

Comment:
I don't think this is required? better-sqlite3 never had node-gyp in its dependencies and from the native modules I know this is not common. Can you link to some docs that recommend doing it? I think this will cause more trouble down the line and it's technically not a dependency but part of the build tooling you need (just like Python, make or gcc). If anything it would be a `optionalDependencies` or `peerDependencies`.

I'm just trying to improve the chances of getting this merged anytime soon.

Reply:
The issue here was, that prebuild is still using node-gyp 6.0.1. If you try to build for electron 20, using this _old_ version, the build would fail, stating that node-gyp below version 8.4.1 is not supported anymore. So the explicit use of at least node-gyp 8.4.1 is required.

In addition to that I only had msvc 2022 available on the machine I was testing on, which requires node-gyp > 9.0.0 as otherwise no compiler is found.

Reply:
I tried using a global installation of node-gyp on a fresh VM. While it now works to run the build-release script, I still encounter this error when trying to run prebuild.

Running `npx --no-install prebuild -r electron -t 20.0.0` yields:

```
prebuild\electron\20.0.0\include\node\node.h(27): fatal error C1189: #error:  "It looks like you are building this native module without using the right config.gypi.  Thi
s normally means that you need to update electron-rebuild (>=3.2.8) or node-gyp (>=8.4.0) if you're building modules directly."
```

I'll remove the dependency, but how to address this issue?

Reply:
Well, maybe we need to include node-gyp https://github.com/prebuild/prebuild/issues/286 and the `overrides` then. I'm not an expert on this, but if it makes the builds more predictable I'm all for it. And I changed my mind on the "it's technically not a dependency", since the `build-release` script literally needs node-gyp when no prebuild is found. But it should be the absolute minimum like `8.4.0` to be as compatible as possible. Otherwise if a project contains a second dependency that depends on node-gyp we're increasing the chances of conflicts. But someone else should weigh in.

Reply:
As sqlite3 is using 8.4.1 I would tend to also go with this version for now. But as I am not that deep into this topic yet, I am open for any good advise.